examples/example-clientlib-ktx-app/app/src/main/java/com/solanamobile/ktxclientsample/viewmodel/SampleViewModel.kt

```diff
@@ -3,41 +3,172 @@ package com.solanamobile.ktxclientsample.viewmodel
+import com.portto.solana.web3.PublicKey
+import com.portto.solana.web3.SerializeConfig
+import com.portto.solana.web3.Transaction
+import com.portto.solana.web3.programs.MemoProgram
 import com.solana.mobilewalletadapter.clientlib.ActivityResultSender
 import com.solana.mobilewalletadapter.clientlib.MobileWalletAdapter
+import com.solana.mobilewalletadapter.clientlib.RpcCluster
+import com.solanamobile.ktxclientsample.usecase.Connected
+import com.solanamobile.ktxclientsample.usecase.NotConnected
+import com.solanamobile.ktxclientsample.usecase.PersistanceUseCase
+import com.solanamobile.ktxclientsample.usecase.SolanaRpcUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
+import org.bitcoinj.core.Base58
 import javax.inject.Inject
 
 data class SampleViewState(
+    val isLoading: Boolean = false,
     val canTransact: Boolean = false,
-    val solBalance: Double = -1.0,
-    val userAddress: String = ""
+    val solBalance: Double = 0.0,
+    val userAddress: String = "",
+    val memoTx: String = ""
 )
 
+val solanaUri = Uri.parse("https://solana.com")
+val iconUri = Uri.parse("favicon.ico")
+val identityName = "Solana"
+
 @HiltViewModel
 class SampleViewModel @Inject constructor(
-    private val walletAdapter: MobileWalletAdapter
+    private val walletAdapter: MobileWalletAdapter,
+    private val solanaRpcUseCase: SolanaRpcUseCase,
+    private val persistanceUseCase: PersistanceUseCase
 ): ViewModel() {
 
+    private fun SampleViewState.updateViewState() {
+        _state.update { this }
+    }
+
     private val _state = MutableStateFlow(SampleViewState())
 
     val viewState: StateFlow<SampleViewState>
         get() = _state
 
-    fun connectToWallet(sender: ActivityResultSender) {
+    fun loadConnection() {
+        val connection = persistanceUseCase.getWalletConnection()
+        if (connection is Connected) {
+            _state.value.copy(
+                isLoading = true,
+                canTransact = true,
+                userAddress = connection.publickKey.toBase58()
+            ).updateViewState()
+
+            viewModelScope.launch {
+                val balance = solanaRpcUseCase.getBalance(connection.publickKey)
+
+                _state.value.copy(
+                    isLoading = false,
+                    solBalance = balance
+                ).updateViewState()
+            }
+        }
+    }
+
+    fun addFunds(sender: ActivityResultSender) {
         viewModelScope.launch {
-            walletAdapter.transact(sender) {
-                val result = authorize(Uri.parse("https://solana.com"), Uri.parse("favicon.ico"), "Solana")
+            val conn = persistanceUseCase.getWalletConnection()
+
+            val currentConn = walletAdapter.transact(sender) {
+                when (conn) {
```

Comment:
Not handled by either of these cases: when the wallet tells us it's time to authorize fresh.